homeassistant/components/samsungtv/bridge.py

```diff
@@ -299,6 +332,175 @@ def _close_remote(self) -> None:
             LOGGER.debug("Could not establish connection")
 
 
+class SamsungTVEncryptedBridge(SamsungTVBridge):
```

Comment:
There is a lot of duplication between this and `SamsungTVWSBridge`.  Does it make sense to make this a super class of `SamsungTVWSBridge` and override methods?

Reply:
I have created a `SamsungTVWSBaseBridge` class, that is inherited by both encrypted and not-encrypted bridges.
That means that `close_remote`, `is_on` and `device_info` methods are shared.

try_connect / send_commands / get_remote is harder to merge because they use different underlying classes.
Might work with a couple generics (_TCommand, _TRemote). I'll try that as a follow-up

Reply:
Well I've implemented the generics. I'm not sure if it's easier to read but I guess it reduces the number of lines a little bit.

Reply:
I put it in, and then I could the generic back out.
I'm happy to put it back in but I think maybe as a follow-up after this one is merged?